app/src/validators/permit.ts

```diff
@@ -0,0 +1,50 @@
+import Joi from 'joi';
+
+import { activityId, uuidv4 } from './common';
+import { validate } from '../middleware/validation';
+import { permitTypeSchema } from './permitType';
+
+const sharedPermitSchema = {
+  permitType: permitTypeSchema,
+  permitTypeId: Joi.number().max(255).required(),
+  activityId: activityId,
+  issuedPermitId: Joi.string().max(255),
+  trackingId: Joi.string().max(255),
+  authStatus: Joi.string().max(255),
+  needed: Joi.string().max(255).required(),
+  status: Joi.string().max(255).required(),
+  submittedDate: Joi.date().iso(),
+  adjudicationDate: Joi.date().iso()
```

Comment:
Optional: Alphabetize attribute ordering, assuming no business reasons on why this ordering is used.